powdr_cli/src/main.rs

```diff
@@ -374,23 +371,46 @@ fn write_params_to_fs(params: &[u8], output_dir: &Path) {
     log::info!("Wrote params.bin.");
 }
 
+fn compile_with_csv_export<T: FieldElement>(
+    file: String,
+    output_directory: String,
+    inputs: String,
+    force: bool,
+    prove_with: Option<Backend>,
+    export_csv: bool,
+    csv_mode: CsvRenderMode,
+) {
+    let result = compile_pil_or_asm::<T>(
+        &file,
+        split_inputs(&inputs),
+        Path::new(&output_directory),
+        force,
+        prove_with,
+    );
+
+    if export_csv {
+        // Compilation result is None if the ASM file file has been compiled before
```

Comment:
This sounds like it caches compilation. Is this really the case? Isn't it just that it returns None if it fails for some reason (one being that it does not overwrite an existing file)?

Reply:
> one being that it does not overwrite an existing file
That's the only reason, and it's what I tried to express here. I'll try to make it clearer.

Reply:
I think it's currently the only reason, but we should abstract over that I would say.